presentations/2024-10-10_what-is-ai-tom/index.qmd

```diff
@@ -0,0 +1,434 @@
+- feed into a convolutional neural network
+- profit?
+:::
+
+## Large corpus of labelled images
+
+### ImageNet
+
+* a large visual database
+* over 14 million hand-annotated images
+* more than 20,000 categories
+* each category has "several hundred" images
+* started in 2006
+
+## Convert the images to tensors
+
+:::: {.columns}
+
+::: {.column width="50%"}
+A fancy way of saying:
+
+turn the images into a 2d table
+
+of values between 0 and 1
+:::
+
+::: {.column width="50%"}
+<div position="relative">
+
+::: { .fragment .fade-in-then-out fragment-index=1}
+<div style="position:absolute; margin-left: auto; margin-right: auto;">
+
+```{r}
+#| fig-width: 5
+#| fig-height: 5
+
+i <- 0
+j <- 0
+
+f <- \(x, y) pmin(1, sqrt((x - 5)^2 + (y - 5)^2) / sqrt(32))
+
+p <- purrr::map(
+  0:10,
+  \(i) purrr::map(
+    0:10,
+    \(j) list(x = i, y = j, v = f(i, j))
+  )
+) |>
+  purrr::flatten() |>
+  dplyr::bind_rows() |>
+  ggplot2::ggplot(ggplot2::aes(x = x, y = y)) +
+  ggplot2::geom_tile(ggplot2::aes(fill = v)) +
+  ggplot2::scale_fill_gradient(low = "black", high = "white") +
+  ggplot2::coord_fixed() +
+  ggplot2::theme_void() +
+  ggplot2::theme(
+    legend.position = "none",
+    plot.background = ggplot2::element_rect(fill = "#f5f4f3", colour = NA)
+  )
+
+p
+```
+
+</div>
+:::
+
+::: { .fragment .fade-in-then-out fragment-index=2}
+<div style="position:absolute; margin-left: auto; margin-right: auto;">
+
+```{r}
+#| fig-width: 5
+#| fig-height: 5
+p + ggplot2::geom_label(ggplot2::aes(label = scales::number(v, 0.01)))
+```
+
+</div>
+:::
+</div>
+:::
+
+::::
+
+## (Convolutional) Neural Networks
+
+{{< video https://youtu.be/aircAruvnKk start="215" width="100%" height="85%" >}}
+
+::: footer
+[3Blue1Brown, YouTube Channel](https://www.youtube.com/c/3blue1brown)
+:::
+
+## Use pre-trained models
+
+Models that have been pre-trained on some image datasets which can be downloaded and used
+
+::: { .incremental }
+* [ResNet](https://huggingface.co/microsoft/resnet-50) (Microsoft Research)
+* [Inception](https://huggingface.co/docs/timm/en/models/inception-v4) (Google)
+* [Trending classifiers from Hugging Face](https://huggingface.co/models?pipeline_tag=image-classification&sort=trending)
+:::
+
+:::{ .fragment }
+Transfer learning is the concept of taking a pre-trained model as a basis, then fine-tuning it to
+classify based on your own images.
+:::
+
+## How can CV be used in Healthcare? { .inverse }
+
+* classification
+* multi-classification
+* object detection
+* event detection
+
+## How can CV be used in Healthcare? { .inverse }
+
+<div style="line-height: 1em">
+::: { .incremental }
+* detecting disease or injury
+* monitoring patients vitals, e.g. respiratory rate
+* detecting bounds of a tumour when planning radiotherapy
+* automating cell counting
+* calculating the grade of a cancer
+* monitor for long-term changes, e.g. AAA
+* devices for patients with vision impairements
```

Comment:
```suggestion
* devices for patients with vision impairments
```

Reply:
ok I need to find out why my gramarly plugin isn't working in vscode :-)